staking/entryWriter.go

```diff
@@ -0,0 +1,196 @@
+package staking
+
+import (
+	"errors"
+	"fmt"
+	"github.com/pegnet/pegnet/spr"
+	log "github.com/sirupsen/logrus"
+	"sync"
+
+	"github.com/FactomProject/factom"
+	"github.com/cenkalti/backoff"
+	"github.com/pegnet/pegnet/common"
+	"github.com/zpatrick/go-config"
+)
+
+type IEntryWriter interface {
+	PopulateECAddress() error
+	NextBlockWriter() IEntryWriter
+	SetSPR(spr *spr.StakingPriceRecord)
+	CollectAndWrite(blocking bool)
+	ECBalance() (int64, error)
+}
+
+// EntryWriter writes the SPRs to factom once the staking is done
+type EntryWriter struct {
+	// We need an spr template to make the entries
+	sprTemplate *spr.StakingPriceRecord
+
+	ec     *factom.ECAddress
+	config *config.Config
+
+	Next *EntryWriter
+
+	EntryWritingFunction func() error
+
+	sync.Mutex
+	sync.Once
+}
+
+func NewEntryWriter(config *config.Config) *EntryWriter {
+	w := new(EntryWriter)
+	w.config = config
+	w.EntryWritingFunction = w.writeStakingRecord
+	return w
+}
+
+func (w *EntryWriter) ECBalance() (int64, error) {
+	return factom.GetECBalance(w.ec.String())
+}
+
+// PopulateECAddress only needs to be called once
+func (w *EntryWriter) PopulateECAddress() error {
+	// Get the Entry Credit Address that we need to write our SPR records.
+	if ecadrStr, err := w.config.String("Miner.ECAddress"); err != nil {
```

Comment:
here error indicates the same behavior for different logic, for non-populated address and address missing in the config, it should be distinguished and maybe forward the explicit error to the user.

Reply:
panic() is just like an exception, it also arises at runtime. Regarding the error, it is processed by panic(err).
https://github.com/dragonheaven/pegnet/blob/feature/pip13/staking/coordinator.go#L39

If you just are going to suggest that we should replace all panic() by alert message to users, it should be done in another pull request.

Exception has been processed like all other errors int this project.
panic() is just enough for this moment.